### ISSUE TITLE
Cannot import Numpy inside a Python Lambda function in sandbox mode

### ISSUE THREAD

**Issue by lpsinger** [architect/architect#1457]
**Describe the issue**
Importing Numpy inside a Python Lambda function results in an error with a stack trace in sandbox mode. The import works fine when you actually deploy it to production, but you can't do any local development.

**Steps to reproduce**
See minimal reproducer here:
https://github.com/nasa-gcn/arc-python-example/tree/e4b0c005d3a796b1079f608f29366e99793ca969

1. Clone
2. Check out that commit hash.
3. Run `npm install`.
4. Run `npm run start`.
5. Navigate to the resulting page in a browser.

**Expected behavior**
You should get a JSON response that says `{"hello": "world"}`.

**Screenshots**
You get the following stack trace:

```
Traceback (most recent call last):
  File "/Users/lpsinger/src/arc-python-example/src/http/get-index/vendor/numpy/core/__init__.py", line 24, in 
    from . import multiarray
  File "/Users/lpsinger/src/arc-python-example/src/http/get-index/vendor/numpy/core/multiarray.py", line 10, in 
    from . import overrides
  File "/Users/lpsinger/src/arc-python-example/src/http/get-index/vendor/numpy/core/overrides.py", line 8, in 
    from numpy.core._multiarray_umath import (
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/lpsinger/src/arc-python-example/src/http/get-index/vendor/numpy/__init__.py", line 130, in 
    from numpy.__config__ import show as show_config
  File "/Users/lpsinger/src/arc-python-example/src/http/get-index/vendor/numpy/__config__.py", line 4, in 
    from numpy.core._multiarray_umath import (
  File "/Users/lpsinger/src/arc-python-example/src/http/get-index/vendor/numpy/core/__init__.py", line 50, in 
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.9 from "/opt/local/bin/python3"
  * The NumPy version is: "1.26.0"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: No module named 'numpy.core._multiarray_umath'


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "", line 29, in 
  File "", line 529, in _check_name_wrapper
  File "", line 1029, in load_module
  File "", line 854, in load_module
  File "", line 274, in _load_module_shim
  File "", line 711, in _load
  File "", line 680, in _load_unlocked
  File "", line 850, in exec_module
  File "", line 228, in _call_with_frames_removed
  File "/Users/lpsinger/src/arc-python-example/src/http/get-index/lambda.py", line 1, in 
    import numpy as np
  File "/Users/lpsinger/src/arc-python-example/src/http/get-index/vendor/numpy/__init__.py", line 135, in 
    raise ImportError(msg) from e
ImportError: Error importing numpy: you should not try to import numpy from
        its source directory; please exit the numpy source tree, and relaunch
        your python interpreter from there.
```

<img width="1211" alt="Screenshot 2023-10-06 at 17 44 57" src="https://github.com/architect/architect/assets/728407/a0dbcbdc-90af-4288-936a-92d9ca8e465d">

**Desktop**
Please complete the following information (if appropriate):
 - OS: macOS
 - Browser Safari
 - Architect version 10.14.2

**Additional context**
[Numpy](https://github.com/numpy/numpy) is _the_ most important Python package for data science, physical sciences, and mathematics, because it is a dependency of almost every other Python package that does any array, vector, or matrix math.

**Comment by lpsinger** [comment]
I see the problem. The hydrated dependencies are for the wrong OS and architecture:

```
$ ls src/http/get-index/vendor/numpy/core/*.so
src/http/get-index/vendor/numpy/core/_multiarray_tests.cpython-39-aarch64-linux-gnu.so
src/http/get-index/vendor/numpy/core/_multiarray_umath.cpython-39-aarch64-linux-gnu.so
src/http/get-index/vendor/numpy/core/_operand_flag_tests.cpython-39-aarch64-linux-gnu.so
src/http/get-index/vendor/numpy/core/_rational_tests.cpython-39-aarch64-linux-gnu.so
src/http/get-index/vendor/numpy/core/_simd.cpython-39-aarch64-linux-gnu.so
src/http/get-index/vendor/numpy/core/_struct_ufunc_tests.cpython-39-aarch64-linux-gnu.so
src/http/get-index/vendor/numpy/core/_umath_tests.cpython-39-aarch64-linux-gnu.so
```

It installed wheels there are appropriate for Linux aarch64, which is what I selected for the Lambda configuration in `app.arc`. However I am using sandbox mode on a macOS x86_64 machine.

In sandbox mode, it should install dependencies that are appropriate for the host OS and architecture, not the target Lambda OS and architecture.

**Comment by ryanblock** [comment]
Sandbox generally does not install deps; in fact, it now efforts to destroy vendor dirs left over from the last deployment. In the case of Python, it prefers to use what's in your Py path, which should be system-specific wheels/eggs etc. What version Sandbox / Arc are you using?

**Comment by lpsinger** [comment]
```
$ npm ls @architect/sandbox
arc-python-example@0.0.0 /Users/lpsinger/src/arc-python-example
└─┬ @architect/architect@10.14.2
  └── @architect/sandbox@5.8.2
```

The problem is that if you have a per-lambda requirements.txt file, then sandbox _does_ hydrate it, and it hydrates it incorrectly.

**Comment by ryanblock** [comment]
That's right, individual requirements.txt and package.json files do cause dep installs. The arch should be system specific for Python, so that would be a bug, although we now expect folks to use Python Lambda treeshaking. Does removing the requirements file from the Lambda resolve the problem?

**Comment by lpsinger** [comment]
Yes, it does, but I would like to use per-Lambda requirements.txt because it suits my workflow better.

**Comment by ryanblock** [comment]
Shipped in 10.15!
